crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3715,9 +3759,30 @@ pub struct BoundMethodType<'db> {
     self_instance: Type<'db>,
 }
 
+/// This type represents a general callable type that are used to represent `typing.Callable`
+/// and `lambda` expressions.
+#[salsa::interned]
+pub struct GeneralCallableType<'db> {
+    signature: Signature<'db>,
```

Comment:
```suggestion
		#[return_ref]
    signature: Signature<'db>,
```

We plan to change the default to return a reference to make this less of a footgun but we have to add `return_ref` annotations for now for all fields that aren't `Copy`